src/listeners/application-commands/chat-input/CorePreChatInputCommandRun.ts

```diff
@@ -12,15 +12,15 @@ export class CoreListener extends Listener<typeof Events.PreChatInputCommandRun>
 
 		// Run global preconditions:
 		const globalResult = await this.container.stores.get('preconditions').chatInputRun(interaction, command, payload as any);
-		if (!globalResult.success) {
-			this.container.client.emit(Events.ChatInputCommandDenied, globalResult.error, payload);
+		if (globalResult.isErr()) {
+			this.container.client.emit(Events.ChatInputCommandDenied, globalResult.unwrapErr(), payload);
 			return;
 		}
 
 		// Run command-specific preconditions:
 		const localResult = await command.preconditions.chatInputRun(interaction, command, payload as any);
-		if (!localResult.success) {
-			this.container.client.emit(Events.ChatInputCommandDenied, localResult.error, payload);
+		if (localResult.isErr()) {
```

Comment:
Shouldn't this use localResult.mapError()? Same for globalResult? Consistency 🙏

Reply:
Adding this here as well as on Discord, no can do because the early returns would drop out that way.

![image](https://user-images.githubusercontent.com/4019718/182940837-7de5d0dd-d1c7-4fa4-9542-86f2bab26d37.png)